src/main/java/frc/robot/subsystem/ElevatorSubsystem.java

```diff
@@ -0,0 +1,366 @@
+    }
+
+    public void adjustWristSmartPosition(ElevatorPosition position) {
+        wristSmartMotionPid.setPositionReference(position.wristAngle);
+    }
+
+    public void adjustSmartPosition(ElevatorPosition position) {
+        switch (state) {
+            case NORMAL:
+            default:
+                if (position.equals(ElevatorSetpointManager.getInstance().getSetpoint(ElevatorSetpoint.INTAKE))
+                        && atPosition(position)) {
+                    disableWristReverseLimit();
+                    new ScheduleCommand(tiltback);
+                    state = WristState.HOLDING;
+                } else if (!atHeight(position) && !stowedWrist(position)) stowWrist();
+                else if (stowedWrist(position)) adjustElevatorSmartPosition(position);
+                else adjustWristSmartPosition(position);
+                break;
+            case HOLDING:
+                if (!tiltback.isScheduled()) holdWristIn();
+                if (!atHeight(position)) {
+                    if (tiltback.isScheduled()) tiltback.cancel();
+                    state = WristState.RETURNING;
+                }
+                break;
+            case RETURNING:
+                stowWrist();
+                if (stowedWrist(position)) {
+                    enableWristReverseLimit();
+                    state = WristState.NORMAL;
+                }
+                break;
+        }
+    }
+
+    public void stowWrist() {
+        wristSmartMotionPid.setPositionReference(Units.Rotations.zero());
+    }
+
+    public void tiltBack() {
+        elevatorWrist.setVoltage(-MoPrefs.elevatorWristPower.get().in(Units.Volts));
+    }
+
+    public void holdWristIn() {
+        elevatorWrist.setVoltage(MoPrefs.elevatorWristHoldPower.get().in(Units.Volts));
+    }
+
+    public boolean atPosition(ElevatorPosition position) {
+        double thresh = MoPrefs.elevatorSetpointVarianceThreshold.get().in(Units.Value);
+        return elevatorRelEncoder.getPosition().isNear(position.elevatorDistance(), thresh)
+                && wristRelEncoder.getPosition().isNear(position.wristAngle(), thresh);
+    }
+
+    public boolean atHeight(ElevatorPosition position) {
+        double thresh = MoPrefs.elevatorSetpointVarianceThreshold.get().in(Units.Value);
+        return elevatorRelEncoder.getPosition().isNear(position.elevatorDistance(), thresh);
+    }
+
+    public boolean stowedWrist(ElevatorPosition position) {
+        double thresh = MoPrefs.elevatorSetpointVarianceThreshold.get().in(Units.Value);
+        return wristRelEncoder.getPosition().isNear(Units.Inches.of(0), thresh);
+    }
+
+    public double getElevatorDirection() {
+        return elevatorSmartMotionPid.getSetpoint();
+    }
+
+    public Current getWristCurrent() {
+        double current = elevatorWrist.getOutputCurrent();
+        wristCurrent.mut_replace(current, Units.Amps);
+        return wristCurrent;
```

Comment:
Can be consolidated into one line
```suggestion
        return wristCurrent.mut_replace(elevatorWrist.getOutputCurrent(), Units.Amps);
```